novawallet/Modules/AssetsSearch/Swaps/SwapAssetsOperationInteractor.swift

```diff
@@ -0,0 +1,123 @@
+import BigInt
+import RobinHood
+
+final class SwapAssetsOperationInteractor: AnyCancellableCleaning {
+    weak var presenter: SwapAssetsOperationPresenterProtocol?
+
+    let stateObservable: AssetListModelObservable
+    let logger: LoggerProtocol
+    let chainAsset: ChainAsset?
+    let assetConversionOperationFactory: AssetConversionOperationFactoryProtocol
+
+    private let operationQueue: OperationQueue
+    private var builder: AssetSearchBuilder?
+    private var directionsCall: CancellableCall?
+    private var availableDirections: [ChainAssetId: Set<ChainAssetId>]?
+
+    init(
+        stateObservable: AssetListModelObservable,
```

Comment:
Like the idea, however this means that we will need stateObservable to go through SwapSetup module.